File: Hechinger.FSK.Web/FSKClient/src/app/services/result/result-builder.ts
import { Result } from "../../models/generated";

export class ResultBuilder {
  private readonly result: Result;

  constructor() {
    this.result = {
      isSuccess: false,
      entities: null,
      errors: new Array<string>(),
      message: '',
    };
  }

  setSuccess(isSuccess: boolean): ResultBuilder {
    this.result.isSuccess = isSuccess;
    return this;
  }
  setMessage(message: string): ResultBuilder {
    this.result.message = message;
    return this;
  }
  setErrors(errors: Array<string>): ResultBuilder {
    this.result.errors = errors;
    return this;
  }
  setEntities(entities: any): ResultBuilder {
    this.result.entities = entities;
    return this;
  }
 
  build(): Result {
    return this.result;
  }
}

File: ProductionControlSuite.FSK.Application/Common/Builders/ResultBuilder.cs
namespace ProductionControlSuite.FSK.Application.Common.Builders
{

    public class ResultBuilder<T>
    {
        private readonly Result<T> result;
        public ResultBuilder()
        {
            this.result = new Result<T>();
        }
        public ResultBuilder<T> SetIsSuccess(bool isSuccess)
        {
            this.result.IsSuccess = isSuccess;
            return this;
        }
        public ResultBuilder<T> SetMessage(string message)
        {
            result.Message = message;
            return this;
        }
        public ResultBuilder<T> AddError(string error)
        {
            if (!this.result.Errors.Any()) result.Errors = new List<string>();
            this.result.Errors.Add(error);
            return this;
        }
        public Result<T> Build() => this.result;
    }
}
